src/Akka.Streams.Kafka.Tests/Integration/PlainSourceIntegrationTests.cs

```diff
@@ -148,16 +148,23 @@ public async Task PlainSource_should_stop_on_deserialization_errors()
 
             var error = probe.Request(elementsCount).ExpectEvent(TimeSpan.FromSeconds(10));
             error.Should().BeOfType<TestSubscriber.OnError>();
-            ((TestSubscriber.OnError)error).Cause.Should().BeOfType<SerializationException>();
+            ((TestSubscriber.OnError)error).Cause.Should().BeOfType<ConsumeException>();
             probe.Cancel();
         }
 
         [Fact]
         public async Task PlainSource_should_resume_on_deserialization_errors()
         {
-            Directive Decider(Exception cause) => cause is SerializationException
-                ? Directive.Resume
-                : Directive.Stop;
+            var callCount = 0;
+            Directive Decider(Exception cause)
+            {
+                if(cause is ConsumeException ex && ex.Error.IsSerializationError())
```

Comment:
So when there's a deserialization error on consume, do we still get a `ConsumeResult` that we could pass downstream?

Reply:
That way, if someone wanted to commit / discard the bad message and advance the consumer - how would they go about doing that? This is something we'd need to have an example of in our code / documentation.

Reply:
Might want to mention schema registry as a general preventative solution to this here.

Reply:
We can't pass the failed de/serialized message downstream because it is of a different type compared to the stream type.
When a message failed to be de/serialized, it is passed inside the ConsumeException as a ConsumeResult<byte[], byte[]>. The only way to examine these messages is by catching it in the Supervision decider or to create the stream without any key/value de/serializer by making it a stream of <byte[], byte[]> instead.

Reply:
To ignore/discard failed de/serialized messages, the user can simply use a decider that always resumes the stream when a de/serialization error occured.

Reply:
> To ignore/discard failed de/serialized messages, the user can simply use a decider that always resumes the stream when a de/serialization error occured.

Still need to commit the offset somewhere - so that will need to be documented for stages that don't automatically commit offsets i.e. `CommitableSource`

Reply:
Offsets are stored in the kafka consumer client, the de/serialization error is the only special case where the actor is not killed when it happened. Setting the decider to resume will automatically skip the failed message.